src/Sound/Tidal/UI.hs

```diff
@@ -1948,8 +1948,11 @@ __binary n num = map (testBit num) $ reverse [0 .. n-1]
 _binary :: Data.Bits.Bits b => Int -> b -> Pattern Bool
 _binary n num = listToPat $ __binary n num
 
-binaryN :: Int -> Pattern Int -> Pattern Bool
-binaryN n p = squeezeJoin $ _binary n <$> p
+_binaryN :: Int -> Pattern Int -> Pattern Bool
+_binaryN n p = squeezeJoin $ _binary n <$> p
+
+binaryN :: Pattern Int -> Pattern Int -> Pattern Bool
+binaryN n p = squeezeJoin $ (\np -> _binaryN np p) <$> n
```

Comment:
I'd normally use `innerJoin` rather than `squeezeJoin`, using the tParam shortcut:
```
 binaryN = tParam _binaryN
```
I think `squeezeJoin` makes sense for the `_binaryN` definition though.

Reply:
Yes, thanks for the feedback, I'm trying to learn the whole `join` stuff. 
Did the fix